src/Button/index.jsx

```diff
@@ -4,40 +4,74 @@ import PropTypes from 'prop-types';
 
 import styles from './Button.scss';
 
-function Button(props) {
-  const {
-    buttonType,
-    className,
-    label,
-    inputRef,
-    isClose,
-    onBlur,
-    onClick,
-    onKeyDown,
-    type,
-    ...other
-  } = props;
+class Button extends React.Component {
+  constructor(props) {
+    super(props);
 
-  return (
-    <button
-      className={classNames([
-        ...className,
-        styles.btn,
-      ], {
-        [styles[`btn-${buttonType}`]]: buttonType !== undefined,
-      }, {
-        [styles.close]: isClose,
-      })}
-      onBlur={onBlur}
-      onClick={onClick}
-      onKeyDown={onKeyDown}
-      type={type}
-      ref={inputRef}
-      {...other}
-    >
-      {label}
-    </button>
-  );
+    const {
+      onBlur,
+      onKeyDown,
+    } = props;
+
+    this.onBlur = onBlur.bind(this);
+    this.onKeyDown = onKeyDown.bind(this);
+
+    this.onClick = this.onClick.bind(this);
+    this.setRefs = this.setRefs.bind(this);
+  }
+
+  /*
+    The button component is given focus explicitly in its onClick to account
+    for the fact that an HTML <button> element in Firefox and Safari does not get
+    focus on onClick.
+
+    See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button.
+  */
+  onClick(e) {
+    this.buttonRef.focus();
+    this.props.onClick(e);
+  }
+
+  setRefs(input) {
+    this.props.inputRef(input);
+    this.buttonRef = input;
+  }
+
+  render() {
+    const {
+      buttonType,
+      className,
+      label,
+      inputRef,
+      isClose,
+      onBlur,
+      onClick,
+      onKeyDown,
+      type,
+      ...other
+    } = this.props;
+
+    return (
+      <button
+        className={classNames([
+          ...className,
+          styles.btn,
+        ], {
+          [styles[`btn-${buttonType}`]]: buttonType !== undefined,
+        }, {
+          [styles.close]: isClose,
+        })}
+        onBlur={onBlur}
+        onClick={onClick}
```

Comment:
should these be `this.onBlur`, `this.onClick`, and `this.onKeyDown`?

Reply:
Yeah, they should be! I forgot to push up my changes last week. Oops. I'll do that now.

Reply:
oh no worries, sorry for the notification spam if this is still a WIP